### PR TITLE
set up supertest, added route tests, created class method

### PR DESCRIPTION
**Pull request by hgranthorner**

### CODE REVIEW

File: server/db/index.js [hunk 1]
```diff
@@ -15,37 +15,40 @@ function getRandom(type, max) {
...
-          .find(chal => chal.css.includes('circle'))
-          .update({ userId: getRandom(users, 3), challengeId: 1 }),
-        userchallenges
-          .find(chal => chal.css.includes('square'))
-          .update({ userId: getRandom(users, 3), challengeId: 2 }),
-        solutions.find(sol => sol.css.includes('circle')).update({ challengeId: 1 }),
-        solutions.find(sol => sol.css.includes('square')).update({ challengeId: 2 }),
+  return (
+    conn
+      .sync({ force: true })
```
Comment (leafoflegend) [inline server/db/index.js]: No app should always drop its database.

File: server/db/index.js [hunk 2]
```diff
@@ -15,37 +15,40 @@ function getRandom(type, max) {
...
+      .then(() => {
+        return Promise.all([
```
Comment (leafoflegend) [inline server/db/index.js]: Nice.

File: server/db/index.js [hunk 3]
```diff
@@ -15,37 +15,40 @@ function getRandom(type, max) {
...
-    .catch(err => console.log(err));
+          images.find(img => img.connector === 'challenge-1').update({ challengeId: 1 }),
+          images.find(img => img.connector === 'userchallenge-1').update({ userchallengeId: 1 }),
+          images.find(img => img.connector === 'challenge-2').update({ challengeId: 2 }),
+          images.find(img => img.connector === 'userchallenge-2').update({ userchallengeId: 2 }),
+        ]);
+      })
+      // eslint-disable-next-line no-console
+      .catch(err => console.error(err))
+  );
```
Comment (leafoflegend) [inline server/db/index.js]: In a closer to production app, we should throw an err.

File: server/db/models/index.js [hunk 4]
```diff
@@ -21,6 +21,30 @@ Image.belongsTo(Userchallenge);
 Challenge.hasMany(Solution);
 Solution.belongsTo(Challenge);
 
+Userchallenge.getActiveAnswer = (userId, challengeId) => {
+  return Userchallenge.findOne({
+    where: { userId, challengeId, submitted: false },
+    include: [Image],
+  }).then(async entity => {
```
Comment (leafoflegend) [inline server/db/models/index.js]: Why is this `async`?

File: server/db/index.js (hunk 2 above)
Comment (hgranthorner) [reply server/db/index.js]: I'd love to take credit for this but this was all Theo's work, I just changed the indentation

File: server/db/models/index.js (hunk 4 above)
Comment (hgranthorner) [reply server/db/models/index.js]: Unsure, will fix

**Pull request merged**